odoo/models.py

```diff
@@ -5123,6 +5123,25 @@ def copy(self, default=None):
 
         return record_copy
 
+    @api.returns(None, lambda value: value.ids)
+    def copy_multi(self, default=None):
+        """ copy_multi(default=None)
+
+        Duplicate records in ``self`` updating it with default values
+
+        :param dict default: dictionary of field values to override in the
+               original values of the copied records, e.g: ``{'field_name': overridden_value, ...}``
+        :returns: new records
+
+        """
+        record_copy_ids = self.env[self._name]
+        for model in self:
+            res = model.copy(default)
+            if res:
+                record_copy_ids += res
+
+        return record_copy_ids
```

Comment:
Your variable names are terrible: what you call `model` is a **record**, and `record_copy_ids` are **not ids**.  And by the way, your method is O(n²) because operator `+` has linear complexity on recordsets...
This will do the job just fine:
```suggestion
        return self.browse([record.copy(default).id for record in self])
```

Reply:
Thx for the feedback 🙏